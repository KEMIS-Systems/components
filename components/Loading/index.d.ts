import React from 'react';
interface ILoading {
    show?: boolean;
}
export default function Loading({ show }: ILoading): React.JSX.Element;
export {};
