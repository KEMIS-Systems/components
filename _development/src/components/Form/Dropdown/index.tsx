import React from 'react';
import {
  Controller,
  FieldValues,
  RegisterOptions,
  FieldPath,
  UseFormReturn,
} from 'react-hook-form';
import { Dropdown as DropdownPrime } from 'primereact/dropdown';
import { classNames } from 'primereact/utils';
import { SelectItemOptionsType } from 'primereact/selectitem';
import { Button as ButtonPrime } from 'primereact/button';

interface IProps<T extends FieldValues> {
  className?: string;
  name: FieldPath<T>;
  label: string;
  options: SelectItemOptionsType;
  selected?: string;
  form: UseFormReturn<T>;
  rules?: RegisterOptions;
  autoFocus?: boolean;
  handleAddButton?: (index: number) => void;
  disabled?: boolean;
}

const Dropdown = <T extends object>({
  className,
  name,
  label,
  form,
  options,
  selected,
  rules,
  autoFocus,
  handleAddButton,
  disabled,
}: IProps<T>) => {
  return (
    <div className={'mb-5 ' + (className !== undefined && className)}>
      {form && (
        <Controller
          name={name}
          control={form.control}
          rules={rules}
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          render={({ field: { ref, ...field }, fieldState }) => {
            return (
              <>
                <label
                  htmlFor={field.name}
                  className={
                    classNames({ 'text-red-400 ': fieldState.error }) + 'block'
                  }
                >
                  {label}
                </label>
                <div className={`${handleAddButton && 'p-inputgroup'}`}>
                  <DropdownPrime
                    id={field.name}
                    options={options}
                    autoFocus={autoFocus}
                    showClear
                    filter
                    className={
                      classNames({ 'p-invalid ': fieldState.error }) + 'w-full '
                    }
                    placeholder={selected}
                    {...field}
                    // ref={ref}
                    onChange={(event) => field.onChange(event.target.value)}
                  />
                  {handleAddButton && (
                    <ButtonPrime
                      type="button"
                      icon="pi pi-plus"
                      className="p-button-success"
                      disabled={disabled}
                      onClick={() => handleAddButton(4)}
                    />
                  )}
                </div>
              </>
            );
          }}
        />
      )}
    </div>
  );
};

export default Dropdown;
