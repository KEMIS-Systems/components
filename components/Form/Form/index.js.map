{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../_development/src/components/Form/Form/index.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,+CAAgE;AAEhE,8DAA+B;AAC/B,iEAAyC;AACzC,4DAAoC;AAoBpC;;;;;;;;;;;;GAYG;AACH,MAAM,IAAI,GAAG,CAAmB,EAC9B,GAAG,EACH,MAAM,EACN,QAAQ,EACR,IAAI,EACJ,MAAM,EACN,cAAc,EACd,QAAQ,EACR,WAAW,EACX,IAAI,EACJ,QAAQ,GACE,EAAE,EAAE;IACd,MAAM,CAAC,WAAW,EAAE,cAAc,CAAC,GAAG,IAAA,gBAAQ,EAAC,KAAK,CAAC,CAAC;IACtD,MAAM,CAAC,SAAS,EAAE,YAAY,CAAC,GAAG,IAAA,gBAAQ,EAAC,KAAK,CAAC,CAAC;IAElD,MAAM,UAAU,GAAG,IAAA,mBAAW,EAAC,GAAG,EAAE;QAClC,IAAI,CAAC,KAAK,EAAE,CAAC;QACb,MAAM,IAAI,MAAM,EAAE,CAAC;IACrB,CAAC,EAAE,CAAC,MAAM,EAAE,IAAI,CAAC,CAAC,CAAC;IAEnB,MAAM,gBAAgB,GAAG,IAAA,mBAAW,EAClC,CAAO,IAAiB,EAAE,EAAE;QAC1B,cAAc,CAAC,IAAI,CAAC,CAAC;QACrB,IAAI,GAAG,EAAE;YACP,IAAI;gBACF,MAAM,QAAQ,GAA2B,WAAW;oBAClD,CAAC,CAAC,WAAW,CAAC,IAAI,CAAC;oBACnB,CAAC,CAAC,IAAI,CAAC;gBACT,IAAI,QAAQ,aAAR,QAAQ,uBAAR,QAAQ,CAAE,EAAE,EAAE;oBAChB,QAAQ,YAAY,QAAQ;wBAC1B,CAAC,CAAC,MAAM,GAAG,CAAC,IAAI,CAAC,GAAG,IAAI,IAAI,QAAQ,CAAC,EAAE,EAAE,EAAE,QAAQ,CAAC;wBACpD,CAAC,CAAC,MAAM,GAAG,CAAC,GAAG,CAAC,GAAG,IAAI,IAAI,QAAQ,CAAC,EAAE,EAAE,EAAE,QAAQ,CAAC,CAAC;oBACtD,eAAK,CAAC,IAAI,CAAC;wBACT,IAAI,EAAE,SAAS;wBACf,KAAK,EAAE,SAAS;qBACjB,CAAC,CAAC;iBACJ;qBAAM;oBACL,MAAM,GAAG,CAAC,IAAI,CAAC,GAAG,IAAI,EAAE,EAAE,QAAQ,CAAC,CAAC;oBACpC,eAAK,CAAC,IAAI,CAAC;wBACT,IAAI,EAAE,SAAS;wBACf,KAAK,EAAE,SAAS;qBACjB,CAAC,CAAC;iBACJ;gBACD,cAAc,IAAI,cAAc,CAAC,IAAI,CAAC,CAAC;gBACvC,UAAU,EAAE,CAAC;aACd;YAAC,OAAO,KAAK,EAAE;gBACd,qBAAI,CAAC,IAAI,CAAC;oBACR,KAAK,EAAE,SAAS;oBAChB,IAAI,EAAE,OAAO;oBACb,IAAI,EAAE,OAAO;oBACb,QAAQ,EAAE,CAAC,KAAK,EAAE,EAAE;wBAClB,IAAI,KAAK,CAAC,aAAa,EAAE;4BACvB,KAAK,CAAC,aAAa,CAAC,KAAK,CAAC,MAAM,GAAG,MAAM,CAAC;yBAC3C;oBACH,CAAC;iBACF,CAAC,CAAC;aACJ;oBAAS;gBACR,cAAc,CAAC,KAAK,CAAC,CAAC;aACvB;SACF;;YACC,OAAO,CAAC,KAAK,CACX,iFAAiF,CAClF,CAAC;IACN,CAAC,CAAA,EACD,CAAC,QAAQ,EAAE,IAAI,EAAE,WAAW,EAAE,UAAU,EAAE,cAAc,CAAC,CAC1D,CAAC;IAEF,IAAA,iBAAS,EAAC,GAAG,EAAE;QACb,MAAM,IAAI,YAAY,CAAC,MAAM,CAAC,CAAC;IACjC,CAAC,EAAE,CAAC,MAAM,CAAC,CAAC,CAAC;IAEb,IAAA,iBAAS,EAAC,GAAG,EAAE;QACb,SAAS,IAAI,IAAI,CAAC,YAAY,CAAC,QAAQ,IAAI,gBAAgB,CAAC,EAAE,CAAC;QAC/D,YAAY,CAAC,KAAK,CAAC,CAAC;IACtB,CAAC,EAAE,CAAC,SAAS,EAAE,IAAI,EAAE,QAAQ,EAAE,gBAAgB,CAAC,CAAC,CAAC;IAElD,OAAO,CACL;QACE,wCACE,IAAI,EAAC,MAAM,EACX,QAAQ,EAAE,IAAI,CAAC,YAAY,CAAC,QAAQ,IAAI,gBAAgB,CAAC,IAExD,QAAQ,CACJ;QACP,8BAAC,iBAAO,IAAC,IAAI,EAAE,WAAW,GAAI,CAC7B,CACJ,CAAC;AACJ,CAAC,CAAC;AAEF,kBAAe,IAAI,CAAC"}