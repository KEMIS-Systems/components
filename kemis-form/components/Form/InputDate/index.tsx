import React from 'react';
import {
  Controller,
  FieldValues,
  RegisterOptions,
  FieldPath,
  UseFormReturn,
} from 'react-hook-form';
import { classNames } from 'primereact/utils';
import { Calendar as CalendarPrime } from 'primereact/calendar';

interface IProps<T extends FieldValues> {
  className?: string;
  name: FieldPath<T>;
  label: string;
  dateFormat?: string;
  form: UseFormReturn<T>;
  rules?: RegisterOptions;
}

const InputDate = <T extends object>({
  className,
  name,
  label,
  dateFormat,
  form,
  rules,
}: IProps<T>) => {
  return (
    <div className={'mb-5 ' + (className !== undefined && className)}>
      {form && (
        <Controller
          name={name}
          control={form.control}
          rules={rules}
          // eslint-disable-next-line @typescript-eslint/no-unused-vars
          render={({ field: { ref, ...field }, fieldState }) => {
            return (
              <>
                <label
                  htmlFor={field.name}
                  className={
                    classNames({ 'text-red-400 ': fieldState.error }) + 'block'
                  }
                >
                  {label}
                </label>
                <CalendarPrime
                  id={field.name}
                  dateFormat={dateFormat || 'dd/mm/yy'}
                  mask="99/99/9999"
                  showIcon
                  showButtonBar
                  showOnFocus={false}
                  className={
                    classNames({ 'p-invalid ': fieldState.error }) + 'w-full'
                  }
                  {...field}
                />
              </>
            );
          }}
        />
      )}
    </div>
  );
};

export default InputDate;
