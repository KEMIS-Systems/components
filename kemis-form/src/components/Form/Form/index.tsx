import React, { useCallback, useEffect, useState } from "react";
import { FieldValues, UseFormReturn } from "react-hook-form";
import Loading from "~/components/Loading";
import api from "~/services/api";
import Toast from "~/utils/toast";
import Swal from "sweetalert2";
import { useLocation } from "react-router-dom";
// import { useRouter } from "next/router";
// import { useLanguage } from '~/hooks/Language';

type K = {
  id?: number;
};

interface IProps<T extends FieldValues> {
  dataEdit?: T & K;
  url?: string;
  submit?: boolean;
  onHide?: () => void;
  onRefreshTable?: (refreshTable: boolean) => void;
  onSubmit?: (data: T) => void;
  getFormData?: (data: FieldValues) => FieldValues | FormData;
  form: UseFormReturn<T>;
  children: React.ReactNode;
}

const Form = <T extends object>({
  onHide,
  dataEdit,
  url,
  submit,
  onRefreshTable,
  onSubmit,
  getFormData,
  form,
  children,
}: IProps<T>) => {
  const location = useLocation();
  // const { language } = useLanguage();
  const [showLoading, setShowLoading] = useState(false);
  const [submitted, setSubmitted] = useState(false);

  const handleHide = useCallback(() => {
    form.reset();
    onHide && onHide();
  }, [onHide, form]);

  const handleSubmitData = useCallback(
    async (data: FieldValues) => {
      setShowLoading(true);
      try {
        const formData: FieldValues | FormData = getFormData
          ? getFormData(data)
          : data;
        if (dataEdit?.id) {
          formData instanceof FormData
            ? await api.post(
                `${url || location.pathname}/${dataEdit.id}`,
                formData
              )
            : await api.put(
                `${url || location.pathname}/${dataEdit.id}`,
                formData
              );
          Toast.fire({
            icon: "success",
            title: "Success",
          });
        } else {
          await api.post(`${url || location.pathname}`, formData);
          Toast.fire({
            icon: "success",
            title: "Success",
          });
        }
        onRefreshTable && onRefreshTable(true);
        handleHide();
      } catch (error) {
        Swal.fire("Opss...", "Error", "error");
      } finally {
        setShowLoading(false);
      }
    },
    [dataEdit, url, location, getFormData, handleHide, onRefreshTable]
  );

  useEffect(() => {
    submit && setSubmitted(submit);
  }, [submit]);

  useEffect(() => {
    submitted && form.handleSubmit(onSubmit || handleSubmitData)();
    setSubmitted(false);
  }, [submitted, form, onSubmit, handleSubmitData]);

  return (
    <>
      <form onSubmit={form.handleSubmit(onSubmit || handleSubmitData)}>
        {children}
      </form>
      <Loading show={showLoading} />
    </>
  );
};

export default Form;
