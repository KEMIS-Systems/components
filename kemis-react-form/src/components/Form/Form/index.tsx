import React, { useCallback, useEffect, useState } from "react";
import { FieldValues, UseFormReturn } from "react-hook-form";
import Swal from "sweetalert2";
import Toast from "../../../utils/toast";
import api from "../../../services/api";
import Loading from "../../Loading";

type K = {
  id?: number;
};

interface IProps<T extends FieldValues> {
  dataEdit?: T & K;
  url?: string;
  submit?: boolean;
  onHide?: () => void;
  onRefreshTable?: (refreshTable: boolean) => void;
  onSubmit?: (data: T) => void;
  getFormData?: (data: FieldValues) => FieldValues | FormData;
  form: UseFormReturn<T>;
  children: React.ReactNode;
}

const Form = <T extends object>({
  onHide,
  dataEdit,
  url,
  submit,
  onRefreshTable,
  onSubmit,
  getFormData,
  form,
  children,
}: IProps<T>) => {
  const [showLoading, setShowLoading] = useState(false);
  const [submitted, setSubmitted] = useState(false);

  const handleHide = useCallback(() => {
    form.reset();
    onHide && onHide();
  }, [onHide, form]);

  const handleSubmitData = useCallback(
    async (data: FieldValues) => {
      setShowLoading(true);
      try {
        const formData: FieldValues | FormData = getFormData
          ? getFormData(data)
          : data;
        if (dataEdit?.id) {
          formData instanceof FormData
            ? await api.post(
                `${url || window.location.pathname}/${dataEdit.id}`,
                formData
              )
            : await api.put(
                `${url || window.location.pathname}/${dataEdit.id}`,
                formData
              );
          Toast.fire({
            icon: "success",
            title: "Success",
          });
        } else {
          await api.post(`${url || location.pathname}`, formData);
          Toast.fire({
            icon: "success",
            title: "Success",
          });
        }
        onRefreshTable && onRefreshTable(true);
        handleHide();
      } catch (error) {
        Swal.fire("Opss...", "Error", "error");
      } finally {
        setShowLoading(false);
      }
    },
    [dataEdit, url, location, getFormData, handleHide, onRefreshTable]
  );

  useEffect(() => {
    submit && setSubmitted(submit);
  }, [submit]);

  useEffect(() => {
    submitted && form.handleSubmit(onSubmit || handleSubmitData)();
    setSubmitted(false);
  }, [submitted, form, onSubmit, handleSubmitData]);

  return (
    <>
      <form onSubmit={form.handleSubmit(onSubmit || handleSubmitData)}>
        {children}
      </form>
      <Loading show={showLoading} />
    </>
  );
};

export default Form;
