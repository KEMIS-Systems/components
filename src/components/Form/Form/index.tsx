import React, { useCallback, useEffect, useState } from "react";
import { FieldValues, UseFormReturn } from "react-hook-form";
import Swal from "sweetalert2";
import Toast from "../../../utils/toast";
import Loading from "../../Loading";
import { AxiosInstance } from "axios";
import { useLanguage } from "../../../hooks/Language";

type K = {
  id?: number;
};

export interface IProps<T extends FieldValues> {
  api?: AxiosInstance;
  dataEdit?: T & K;
  url: string;
  submit?: boolean;
  onHide?: () => void;
  onRefreshTable?: (refreshTable: boolean) => void;
  onSubmit?: (data: T) => void;
  getFormData?: (data: FieldValues) => FieldValues | FormData;
  form: UseFormReturn<T>;
  children: React.ReactNode;
}

/**
 * Displays a form with the specified fields. By default when submit it POST/PUT the data in the specified url.
 *
 * @children This component Must have a child element (ej: fields)
 * @param form to control the form data
 * @param onHide (optional) callback to control what happen when you close/restart/clean the form
 * @param api (optional) allows to make the POST/PUT request to our service/api
 * @param url (optional) path to POST/PUT our form data. Previusly you must include the 'api' property and specify your 'base_url' of the service you want to do the request.
 * @param dataEdit (optional) obj that include initial data to show in fields
 * @param onSubmit (optional) callback to change what happens on Submit
 * @param onRefreshTable (optional) callback to refresh data in other site (if needed)
 * @param getFormData (optional) callback which must returns the form data
 */
const Form = <T extends object>({
  api,
  onHide,
  dataEdit,
  url,
  submit,
  onRefreshTable,
  onSubmit,
  getFormData,
  form,
  children,
}: IProps<T>) => {
  const { language } = useLanguage();
  const [showLoading, setShowLoading] = useState<boolean>(false);
  const [submitted, setSubmitted] = useState<boolean>(false);

  useEffect(() => {
    if (dataEdit) {
      form.reset(dataEdit);
    }
  }, [dataEdit, form]);

  const handleHide = useCallback(() => {
    form.reset();
    onHide && onHide();
  }, [onHide, form]);

  const handleSubmitData = useCallback(
    async (data: FieldValues) => {
      setShowLoading(true);
      if (api) {
        try {
          const formData: FieldValues | FormData = getFormData
            ? getFormData(data)
            : data;
          if (dataEdit?.id) {
            formData instanceof FormData
              ? await api.post(`${url}/${dataEdit.id}`, formData)
              : await api.put(`${url}/${dataEdit.id}`, formData);
            await Toast.fire({
              icon: "success",
              title: language.pages.alerts.edit.success,
            });
          } else {
            await api.post(`${url}`, formData);
            await Toast.fire({
              icon: "success",
              title: language.pages.alerts.add.success,
            });
          }
          onRefreshTable && onRefreshTable(true);
          handleHide();
        } catch (error) {
          await Swal.fire({
            title: "Opss...",
            text: "Error",
            icon: "error",
            willOpen: (popup) => {
              if (popup.parentElement) {
                popup.parentElement.style.zIndex = "5000";
              }
            },
          });
        } finally {
          setShowLoading(false);
        }
      } else
        console.error(
          "If you want to POST/PUT, you must include the 'api' property in Form component."
        );
    },
    [dataEdit, url, getFormData, handleHide, onRefreshTable]
  );

  useEffect(() => {
    submit && setSubmitted(submit);
  }, [submit]);

  useEffect(() => {
    submitted && form.handleSubmit(onSubmit || handleSubmitData)();
    setSubmitted(false);
  }, [submitted, form, onSubmit, handleSubmitData]);

  return (
    <>
      <form
        role="form"
        onSubmit={form.handleSubmit(onSubmit || handleSubmitData)}
      >
        {children}
      </form>
      <Loading show={showLoading} />
    </>
  );
};

export default Form;
