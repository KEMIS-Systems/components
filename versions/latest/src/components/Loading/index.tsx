import React from 'react';

import { Container } from './styles';

interface ILoading {
  show?: boolean;
}

export default function Loading({ show }: ILoading) {
  return (
    <>
      {show && (
        <Container className="loading-box">
          <div className="loader" role="status" />
        </Container>
      )}
    </>
  );
}
