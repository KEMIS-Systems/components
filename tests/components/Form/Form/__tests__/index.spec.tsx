import React from "react";
import "../../../../setupTests";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Form from "..";
import axios from "axios";
import { useForm } from "react-hook-form";
import Loading from "../../../Loading";

jest.mock("react-hook-form", () => ({
  useForm: jest.fn(),
}));

describe("Form component", () => {
  const baseUrl = "https://our-backend-base-url.com";
  const path = "/api/form";
  const url = `${baseUrl}${path}`;

  const api = axios.create({
    baseURL: baseUrl,
  });

  const formData = {
    name: "John Doe",
    email: "johndoe@example.com",
    message: "Hello world",
  };
  const dataEdit = { id: 1, ...formData };
  const onRefreshTable = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useForm as jest.Mock).mockReturnValue({
      handleSubmit: jest.fn(),
    });
  });

  test("it renders", () => {
    const onSubmit = jest.fn().mockResolvedValueOnce(undefined);

    render(
      <Form
        api={api}
        dataEdit={dataEdit}
        path={path}
        onRefreshTable={onRefreshTable}
        onSubmit={onSubmit}
        form={useForm()}
      >
        <label htmlFor="name">Name</label>
        <input id="name" name="name" defaultValue={formData.name} />
        <label htmlFor="email">Email</label>
        <input id="email" name="email" />
        <label htmlFor="message">Message</label>
        <textarea id="message" name="message" />
        {/* <button type="submit">Submit</button> */}
      </Form>
    );
  });

  test("update form data successfully", async () => {
    const formDataEdited = {
      name: "John Doe",
      email: "edited_form@example.com",
      message: "Hello world edited",
    };
    const onSubmit = jest.fn().mockResolvedValueOnce(undefined);

    render(
      <Form
        api={api}
        dataEdit={dataEdit}
        path={path}
        onRefreshTable={onRefreshTable}
        form={useForm()}
      >
        <label htmlFor="name">Name</label>
        <input id="name" name="name" value={formDataEdited.name} />
        <label htmlFor="email">Email</label>
        <input id="email" name="email" value={formDataEdited.email} />
        <label htmlFor="message">Message</label>
        <textarea id="message" name="message" value={formDataEdited.message} />
      </Form>
    );

    const nameInput = screen.getByLabelText("Name");
    const emailInput = screen.getByLabelText("Email");
    const messageInput = screen.getByLabelText("Message");

    userEvent.type(nameInput, formDataEdited.name);
    userEvent.type(emailInput, formDataEdited.email);
    userEvent.type(messageInput, formDataEdited.message);

    // Submit the form
    fireEvent.submit(screen.getByRole("form"));

    // Loading component is rendered
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();

    // The PUT request is called with the correct arguments:
    const putSpy = jest.spyOn(api, "put"); // Creates a mock function similar to jest.fn but also tracks calls to object[methodName]
    putSpy.mockResolvedValueOnce(formData);
    expect(putSpy).toHaveBeenCalledWith(`${url}/${dataEdit.id}`, formData);
    // expect(api.put).toHaveBeenCalledWith(`${path}/${dataEdit.id}`, formData);

    expect(onRefreshTable).toHaveBeenCalledWith(true);
    expect(onSubmit).toHaveBeenCalledWith(formData);
  });
});
