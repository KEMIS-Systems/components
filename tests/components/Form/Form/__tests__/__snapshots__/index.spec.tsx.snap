// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component update form data successfully 1`] = `<DocumentFragment />`;
